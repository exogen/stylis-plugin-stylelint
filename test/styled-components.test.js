const React = require('react')
const styled = require('styled-components').default
const TestRenderer = require('react-test-renderer')

describe('using a stylis mock to lint styled-components', () => {
  it('passes lint', () => {
    const Title = styled.h1`
      color: red;
    `
    const wrapper = TestRenderer.create(<Title />)
    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('logs multiple lint errors', () => {
    // prettier-ignore
    const Title = styled.h1`
      background: d2
      color: {}
    `
    const wrapper = TestRenderer.create(<Title />)
    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
