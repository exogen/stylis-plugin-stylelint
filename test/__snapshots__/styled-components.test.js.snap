// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`using a stylis mock to lint styled-components logs multiple lint errors 1`] = `
<h1
  className="styled-componentstest__Title-s6un9az-1 jzxgKV"
/>
`;

exports[`using a stylis mock to lint styled-components passes lint 1`] = `
<h1
  className="styled-componentstest__Title-s6un9az-0 lgCYTL"
/>
`;
